code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -710,6 +710,35 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
 		to_chat(src, "<span class='warning'>[alert_msg]</span>")
 	return success
 
+/datum/AI_Module/large/upgrade_turrets/upgrade(mob/living/silicon/AI/AI)
```

Comment:
🤔

Reply:
This datum already exists in line 390

Reply:
uh what's that doing there..

Reply:
You may wanna do some housekeeping cause there's this datum and the fact you're removing the cleaning tools from the janitorial robot module suggest some things *might* be slipping through